sdks/python/apache_beam/io/textio.py

```diff
@@ -78,6 +78,10 @@ def position(self, value):
                          'size of data %d.', value, len(self._data))
       self._position = value
 
+    def reset(self):
+      self.data = ''
```

Comment:
<!--thread_id:cc_145273015_t; commit:70ade9d4fe6a8e31ff0be4d39b731510c93b4bf9; resolved:1-->
<!--section:context-quote-->
> **mhsaul** wrote:
> The reason for this is that we want to be able to use the same buffer in the _read_records and _process_header methods. Setting read_buffer = ReadBuffer('', 0) won't behave the same in this case since it will only reset the buffer in the scope of the _process_header method but not reset the buffer in the caller of _process_header.

<!--section:body-->
Done.